akka-actor/src/main/scala-2.13+/akka/util/ByteString.scala

```diff
@@ -0,0 +1,1252 @@
+/*
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
```

Comment:
all these versions of ByteString and ByteIterator is worrisome, but I guess it's not much we can do about it (the price we have to pay for the poor choice of extending Vector)

Reply:
Yes it's pretty tricky. in akka-http `Uri.Query` I managed to share the extension of `LinearSeq`, but it's pretty hairy (https://github.com/akka/akka-http/pull/2298/files#diff-e0948a9dfaf145ac83bbbefe9080d77dR584) so I think heer keeping separate implementations is probably the sensible thing to do...